main.go

```diff
@@ -51,6 +48,20 @@ func init() {
 	// +kubebuilder:scaffold:scheme
 }
 
+// getWatchNamespace returns the Namespace the operator should be watching for changes
+func getWatchNamespace() (string, error) {
+	// WatchNamespaceEnvVar is the constant for env variable WATCH_NAMESPACE
+	// which specifies the Namespace to watch.
+	// An empty value means the operator is running with cluster scope.
+	var watchNamespaceEnvVar = "WATCH_NAMESPACE"
+
+	ns, found := os.LookupEnv(watchNamespaceEnvVar)
+	if !found {
+		return "", fmt.Errorf("%s must be set", watchNamespaceEnvVar)
+	}
```

Comment:
What is the consecuence to leave the value of WATCH_NAMESPACE as an empty string, according to the Roles and RolesBindings linked to the service account that run the operator controller? Should it be able to watch the CR across all the cluster if it is provided empty? will it fail to watch the other objects and only the CR objects created into the namespace where the operator is deployed?